Services/COPage/classes/class.ilPCVerificationGUI.php

```diff
@@ -179,24 +207,47 @@ protected function initForm($a_insert = false)
 	}
 
 	/**
-	* Create new verification
-	*/
+	 * Create new verification
+	 * @throws ilException
+	 */
 	function create()
 	{
 		$form = $this->initForm(true);
-		if($form->checkInput())
-		{
-			$type = ilObject::_lookupType($form->getInput("object"));
-			if($type)
-			{			
+		if($form->checkInput()) {
+			$option = $form->getInput('certificate_selection');
+
+			if ('certificate_workspace_option' === $option) {
+				$type = ilObject::_lookupType($form->getInput("object"));
+				if($type) {
+					$this->content_obj = new ilPCVerification($this->getPage());
+					$this->content_obj->create($this->pg_obj, $this->hier_id, $this->pc_id);
+					$verificationObjectId = $form->getInput("object");
+
+					$this->content_obj->setData($type, $verificationObjectId);
+
+					$this->updated = $this->pg_obj->update();
+					if ($this->updated === true) {
+						$this->ctrl->returnToParent($this, "jump".$this->hier_id);
+					}
+				}
+			} elseif ('certificate_persistent_option' === $option) {
+				$objectId = $form->getInput("persistent_object");
+
+				$userId = $this->user->getId();
+
+				$certificateFileService = new ilPortfolioCertificateFileService();
+				$certificateFileService->createCertificateFile($userId, $objectId);
+
 				$this->content_obj = new ilPCVerification($this->getPage());
 				$this->content_obj->create($this->pg_obj, $this->hier_id, $this->pc_id);
-				$this->content_obj->setData($type, $form->getInput("object"));
+				$this->content_obj->setData('crta', $objectId);

```

Comment:
@legionth I think we must use the 'crta' prefix also for the ID here.

Reply:
TBH I actually don't know. The code here is very confusing (references in method header, constructors, afterward setters etc.). As far as I can estimate the check for the type before process with the object id, is unique enough.

Reply:
So I faked some obj_ids in my table to be the same as other objects in the portfolio (e.g. workspace verification object). It seems setting the type `crta` is unique enough :+1: .